### PR TITLE
feat(skills): add dpgen-run skill for concurrent learning workflow

### PR DESCRIPTION
**Pull request by SchrodingersCattt**
## Summary by CodeRabbit

* **Documentation**
  * Added a comprehensive guide for running DP-GEN concurrent-learning workflows, including `param.json` and `machine.json` configuration, setup requirements, runtime environments, validation steps, expected outputs, and configuration guidance.
  * Documented training, exploration, first-principles, descriptor, input-format, machine-configuration, authorization, and option-preservation requirements.

* **Tests**
  * Added checks ensuring documentation links resolve to valid repository resources.
  * Added validation for machine configuration examples, required workflow stages, resource settings, and execution commands.
  * Enabled standalone execution of the documentation validation tests.

### CODE REVIEW

File: skills/dpgen-run/references/workflow.md
Comment (SchrodingersCattt) [inline skills/dpgen-run/references/workflow.md:1]: The name could vary.

File: skills/dpgen-run/references/workflow.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+|-- init_data/
+|   `-- system_000/
+|       |-- type_map.raw
+|       |-- type.raw
+|       `-- set.000/
+|-- assets/
+|   `-- structures/
+|-- cp2k_basis_pp_file/   # only when CP2K needs it
+`-- iter.*/              # created by DP-GEN
+```
```
Comment (SchrodingersCattt) [inline skills/dpgen-run/references/workflow.md:49]: No need to specify CP2K.
Raw workspace could be nested in init_data. e.g. init_data/raw_xxx

File: skills/dpgen-run/references/workflow.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+## Result contract
+
+Before execution, report:
+
+1. absolute paths to `param.json` and `machine.json`
+1. exact launcher command
+1. validation results
+1. unresolved required inputs
+1. expected cost-bearing stages
+
```
Comment (SchrodingersCattt) [inline skills/dpgen-run/references/workflow.md:56]: Should add a monitoring stage to track accuracy evolutions across iterations.

File: skills/dpgen-run/SKILL.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+- The exact launcher is `dpgen run param.json machine.json`.
+- Treat the launcher shell and dispatched stage environments as separate layers.
+- Prepare and validate by default. Execute only when the user explicitly requests it and confirms the exact command after validation.
+
+## Workflow
+
+1. Confirm this is a `dpgen run` task and inspect existing files first.
+1. Collect only missing inputs; patch working configurations instead of rebuilding them.
+1. Load only the references needed for the current task.
+1. Preserve scientific and site-specific choices unless the user approves changes.
```
Comment (SchrodingersCattt) [inline skills/dpgen-run/SKILL.md]: Should there be a list?

File: skills/dpgen-run/SKILL.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+## Progressive references
+
+Read only the relevant files:
+
+- [Workflow policy](references/workflow.md): scope, requirements, missing-input policy, layout, and reporting.
+- [param.json guide](references/param-json.md): systems, training, exploration, FP fields, descriptors, thresholds, and examples.
+- [machine.json guide](references/machine-json.md): stage commands, contexts, resources, scheduler environments, and examples.
+- [Validation and execution](references/validation-and-run.md): schema checks, data checks, authorization, launch, and output inspection.
+
```
Comment (SchrodingersCattt) [inline skills/dpgen-run/SKILL.md:36]: All too long. Compact and leave the most valuable messages.

File: skills/dpgen-run/references/machine-json.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+## Starting profiles
+
+Choose a profile that matches where the outer command runs:
+
+- server-local scheduler: [scheduler example](../../../examples/machine/DeePMD-kit-1.x/machine-lsf-slurm-cp2k.json)
+- pure local shell: [local example](../../../examples/machine/DeePMD-kit-1.x/machine-local.json)
+
+Patch the closest existing working configuration. Do not transplant site-specific paths or scheduler settings blindly.
+
```
Comment (SchrodingersCattt) [inline skills/dpgen-run/references/machine-json.md]: Do not write CP2K-spcific instructions.
DeePMD-kit-1.x is way too out-of-date...

File: skills/dpgen-run/references/validation-and-run.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+`command`, `machine` (`batch_type`, `context_type`, `local_root`, and any
+required `remote_root`/profile), and `resources` (`number_node`, CPU/GPU counts,
+`group_size`, queue/partition flags, and scheduler `source_list`). See the
+[repository machine schema walkthrough](../../../doc/run/example-of-machine.md)
+for generic local, scheduler, and SSH shapes.
+
+The launcher shell and dispatched tasks are different environments. Scheduler
+jobs must activate their own runtime through `source_list`; outer activation is
+not inherited reliably. The training command must select the same backend as
+`train_backend` (for DeePMD-kit 3.x PyTorch this may be `dp --pt`). Validate
```
Comment (SchrodingersCattt) [inline skills/dpgen-run/references/validation-and-run.md]: **Factual accuracy (major)** — DP-GEN appends the backend flag automatically; explicit `--pt` in `machine.json` command is redundant at best, an error at worst.

In `run_train_dp`, `dpgen/generator/run.py` does:
```python
if suffix == ".pth":
    train_command += " --pt"
```
The `suffix` comes from `_get_model_suffix(jdata)`, which maps `train_backend: "pytorch"` → `.pth`. So for a 3.x PyTorch run with `train_backend: "pytorch"`, DP-GEN already emits `dp --pt train input.json ...`. A user who follows this skill and also writes `"command": "dp --pt"` in `machine.json` produces `dp --pt --pt ...`, which deepmd-kit's `main.py` rejects (`--pt` is in an argparse mutually-exclusive group).

Better guidance: keep `command` as `dp`, let `train_backend` drive the flag, and mention that DP-GEN appends `--pt`/`--jax` automatically based on `train_backend`.

File: skills/dpgen-run/references/workflow.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+## Generic layout
+
+```text
+project/
+|-- param.json
+|-- machine.json
+|-- init_data/
+|   `-- raw_xxx/          # DeepMD NumPy system(s)
+|-- assets/               # structures and runtime support files
```
Comment (SchrodingersCattt) [inline skills/dpgen-run/references/workflow.md]: **Accuracy (minor)** — `assets/` is not part of the DP-GEN run project layout. `dpgen run` creates only `iter.*/` (00.train, 01.model_devi, 02.fp) and reads `param.json`/`machine.json`; no code, doc, or example in this repo places user input files under `assets/`. Suggest replacing with a neutral directory name (e.g. `structures/` or `inputs/`) or dropping the line.

File: skills/dpgen-run/SKILL.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+- [Workflow guidance](references/workflow.md): scope, layout, policy, and reporting.
+- [Configuration, validation, and execution](references/validation-and-run.md): both JSON files, generated-input checks, and safe launch/restart checks.
+- [Monitoring and troubleshooting](references/monitoring.md): accuracy trends, stage progress, and evidence-based diagnosis.
+
+## Guardrails
+
+- Never merge the JSON files or invent paths, queues, modules, or commands.
+- Preserve descriptor, backend, thresholds, schedules, ensemble, and `type_map` choices.
+- Spell `se_atten_v2` exactly; `fp_style: "none"` is invalid for current `dpgen run`.
```
Comment (SchrodingersCattt) [inline skills/dpgen-run/SKILL.md]: **Nit / precision** — `se_atten_v2` is valid, but the standalone guardrail may mislead. In current DeePMD-kit, `se_atten` (pre-v3) is the TensorFlow-only legacy descriptor and `se_atten_v2` is registered for the new backends (`deepmd/dpmodel/descriptor/se_atten_v2.py:36`, `deepmd/tf/descriptor/se_atten_v2.py:24`); both are accepted spellings and the exact artifact depends on the installed deepmd-kit/backend. Suggest rewriting as: "Verify the descriptor `type` spelling against the installed DeePMD-kit backend; e.g. `se_atten_v2` is registered in current releases."

File: skills/dpgen-run/references/validation-and-run.md [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+DP-GEN can fill it from `init_data_sys`. Verify `deepmd_version` and the
+installed stack before using version-specific descriptors or features.
+
+`train_backend` accepts `tensorflow` and `pytorch`, but compatibility still
+depends on the installed DeePMD-kit and descriptor. Verify the combination;
+spell `se_atten_v2` exactly. Ordinary LAMMPS exploration uses the default
+`model_devi_engine`; set an alternative only when verified. Current `dpgen run`
+does not accept `fp_style: "none"`.
+
+Each `init_data_sys` directory must contain `type_map.raw`, `type.raw`, and
```
Comment (SchrodingersCattt) [inline skills/dpgen-run/references/validation-and-run.md]: **Accuracy (minor)** — `init_data_sys` may also be an HDF5 file, not only a directory, and `type_map.raw` is not strictly required.

Per `dpgen/dpdata` convention: `init_data_sys` entries can be either a DeepMD NumPy system directory or an HDF5 file (`doc_init_data_sys` in `dpgen/generator/arginfo.py`: "The path can be either a system directory containing NumPy files or an HDF5 file"). In dpdata's `formats/deepmd/raw.py`, `type_map.raw` is optional — `load_type` falls back to the `type_map` argument and then to generated `Type_N` names, and only `type.raw` is required. Suggest: "Each NumPy system directory must contain at least `type.raw` and `set.000/`; `type_map.raw` should match `param.json.type_map` when present. HDF5 init data is also accepted."

File: skills/dpgen-run/references/validation-and-run.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+
+Normalize `param.json` with `run_jdata_arginfo()` and convert `machine.json` with
+`convert_mdata()`. Resolve all data/structure paths, compare type maps, check
+executables and scheduler access, and verify FP inputs and cost limits.
+
+Before submission, inspect the generated `input.json` for every stage/task. Do
+not infer the effective configuration from the root files: check model count,
+seed, checkpoint, data systems, `sys_idx`, temperatures/pressures, and final
+training steps. When reuse or iteration overrides exist, assert both the default
+and effective values.
```
[the remark this answers is not in the document]
Comment (SchrodingersCattt) [reply skills/dpgen-run/references/validation-and-run.md]: Fixed in `b9ad029`: validation now inspects stage-specific artifacts—training `input.json`, LAMMPS `input.lammps`/`job.json`, and backend-specific FP inputs such as VASP `INCAR`—instead of requiring `input.json` for every stage.

File: skills/dpgen-run/references/validation-and-run.md [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+`train_backend` accepts `tensorflow` and `pytorch`, but compatibility still
+depends on the installed DeePMD-kit and descriptor. Verify descriptor spelling
+against that backend; current releases register `se_atten_v2`.
+Ordinary LAMMPS exploration uses the default
+`model_devi_engine`; set an alternative only when verified. Current `dpgen run`
+does not accept `fp_style: "none"`.
+
+Each `init_data_sys` entry may be a DeepMD NumPy system directory or an HDF5
+file. Each NumPy directory must contain at least `type.raw` and `set.000/`.
+When `type_map.raw` is present, it must match `param.json.type_map` exactly.
```
[the remark this answers is not in the document]
Comment (SchrodingersCattt) [reply skills/dpgen-run/references/validation-and-run.md]: Fixed in `b9ad029`: `type_map.raw` is optional; when present, symbols only need to be included in `param.json.type_map`, and subsets/different orders are accepted because DeePMD remaps by element name. Without it, `type.raw` must already use model order.

File: skills/dpgen-run/references/validation-and-run.md [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+
+`train_backend` accepts `tensorflow` and `pytorch`, but compatibility still
+depends on the installed DeePMD-kit and descriptor. Verify descriptor spelling
+against that backend; current releases register `se_atten_v2`.
+Ordinary LAMMPS exploration uses the default
+`model_devi_engine`; set an alternative only when verified. Current `dpgen run`
+does not accept `fp_style: "none"`.
+
+Each `init_data_sys` entry may be a DeepMD NumPy system directory or an HDF5
+file. Each NumPy directory must contain at least `type.raw` and `set.000/`.
```
[the remark this answers is not in the document]
Comment (SchrodingersCattt) [reply skills/dpgen-run/references/validation-and-run.md]: Fixed in `b9ad029`: validation guidance now follows DP-GEN discovery rules, allowing parent directories with multiple systems, HDF5 entries, and any loadable `set.*` directory rather than requiring `set.000` at the input path.
